core/src/test/java/org/apache/iceberg/hadoop/TestCachingCatalog.java

```diff
@@ -276,6 +283,44 @@ public void testCacheExpirationIsDisabledByANegativeValue() throws IOException {
         catalog.isCacheExpirationEnabled());
   }
 
+  @Test
+  @Ignore("reproduces https://github.com/apache/iceberg/issues/3791")
```

Comment:
This test passes with the changes in #3801

Reply:
Eventually, we might want to put this test into the source (and leave it with the `Ignore`) so it can be reran. At the least lets, reference the other PR here so people will have a higher chance of seeing the link: https://github.com/apache/iceberg/pull/3801